### PR TITLE
Sidebar component

### PR DESCRIPTION
**Pull request by foster55f**
What's this PR do?
This PR creates all the Sidebar and SidebarProject. This PR also created a util directory for apiCalls and action creators and reducers for the projects.

Where should the reviewer start?
Sidebar.js and SidebarProject.js. And the actions index.js and projects.js for reducers.

### CODE REVIEW

File: src/Components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -1,13 +1,42 @@
...
+
+  render(){
+    return (
+      <>
+        {this.props.projects.map(project => {
+        return (
+          <SidebarProject
+            title={project.title}
+          />
+        )
```
Comment (trishalanglois) [inline src/Components/Sidebar/Sidebar.js]: Can you change this part to match the comp?  So, Projects will be at the top of the Sidebar, with an underline (you'll need to make a little div for this, then just show the border on one side), then each title will be rendered after that.  The way that you have it now, I think each title will have Projects: [title name].

**Comment by trishalanglois** [comment]
Were you able to do any styling for these, too?  I don't see any CSS files changed, so that's probably why they're being rendered at the top right now.

**Comment by trishalanglois** [comment]
Great job setting up redux!  I'll work through a Hooks/redux component tonight and see if that helps us refactor this one.

File: src/Components/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
-      <SidebarProject />
-    </>
-  )
+
+export class Sidebar extends React.Component {
+  componentDidMount() {
+    retrieveProjects()
+      .then(data => {
+        this.props.addProjects(data)
+      })
```
Comment (trishalanglois) [inline src/Components/Sidebar/Sidebar.js]: Looks like this is what we want to use a hook for in this component.  Check out useEffect Part 1 in the Scrimba course for a walk through on how to refactor this :)
https://scrimba.com/p/p7P5Hd/caqbeZTw

**Comment by foster55f** [comment]
Refactored component to use all hooks.  Added styling

**Pull request merged**